tests/test_main.py

```
@@ -1,431 +1,433 @@
+    )
+
+
+@pytest.mark.parametrize(
+    'kilometers,number,result',
+    [
+        (50, 3, 150),
+        (30, 7, 210),
+        (45, 5, 225),
+        (1.5, 3.0, 4.5),
+        (21.4, 5.88, 125.832)
+    ]
+)
+def test_distance_class_mul_number(kilometers, number, result):
+    distance1 = Distance(kilometers)
+    distance2 = distance1 * number
+    assert isinstance(distance2, Distance), (
+        "Result of Distance instance multiplied by number should be "
+        "Distance instance"
+    )
+    assert distance2.km == result, (
+        f"distance2.km should equal to {result}, "
+        f"when 'distance2 = Distance({kilometers}) * {number}'"
+    )
+
+
+def test_distance_class_mul_distance():
+    distance1 = Distance(5)
+    distance2 = Distance(3)
+    try:
+        result = distance1 * distance2
+    except TypeError:
+        pass
+    else:
+        assert result is None, (
+            "'__mul__' method should not accept Distance instance"
+        )
+
+
+@pytest.mark.parametrize(
+    'kilometers,number,result',
+    [
+        (50, 3, 16.67),
+        (30, 7, 4.29),
+        (45, 5, 9),
+        (12.6, 3.3, 3.82),
+        (26.88, 5.6, 4.8)
+    ]
+)
+def test_distance_class_truediv_number(kilometers, number, result):
+    distance1 = Distance(kilometers)
+    distance2 = distance1 / number
+    assert isinstance(distance2, Distance), (
+        "Result of Distance instance divided by number should be "
+        "Distance instance"
+    )
+    assert distance2.km == result, (
+        f"distance2.km should equal to {result}, "
+        f"when 'distance2 = Distance({kilometers}) / {number}'"
+    )
+
+
+def test_distance_class_truediv_distance():
+    distance1 = Distance(30)
+    distance2 = Distance(3)
+    try:
+        result = distance1 / distance2
+    except TypeError:
+        pass
+    else:
+        assert result is None, (
+            "'__truediv__' method should not accept Distance instance"
+        )

```

This test should use `with pytest.raises(TypeError):` to verify that a `TypeError` is raised when dividing a `Distance` by another `Distance`. The current implementation incorrectly tests that the operation returns `None`, which confirms the incorrect behavior in the `__truediv__` method. The method should raise a `TypeError` for unsupported operands, not return `None`.